docs/advanced/security.rst

```diff
@@ -0,0 +1,52 @@
+Further, some data can be sent encrypted to the control service using the public key of the signed certificate presented by the control service.
+If the control service is able to send the same data back encrypted with the public key presented by the convergence agent's signed certificate, both sides of the communication can be sure that the other holds the private key corresponding to their respective certificates.
+
+Provided that the cluster administrator has kept the private key of the certificate authority a secret (that is, the key file has not been shared or copied anywhere), this identification process confirms that the machine the node is talking to is indeed the genuine control service for the cluster and vice-versa.
+All components of a Flocker cluster identify to each other in this way.
+By providing this identity verification layer, the cluster is able to prevent unauthorised attempts to interact with the control service, convergence agents and REST API.
+
+Security Benefits
+=================
+
+The TLS client certification layer used by Flocker provides a number of security benefits to a cluster:
+
+- Prevents unauthorised requests to the REST API.
+- Prevents `man-in-the-middle <http://en.wikipedia.org/wiki/Man-in-the-middle_attack>`_ and `identity replay <http://en.wikipedia.org/wiki/Replay_attack>`_ attacks between the control service and convergence agents by verifying the identity of cluster components.
+- Securely encrypts communications over the public internet between the control service and convergence agents.
+- Securely encrypts communications between the control service and an API end user.
+- Prevents connections to the control service and convergence agents from unidentified or unauthorised sources.
+- Prevents communications between API users, the control service and convergence agents from being altered or tampered with in transit over the public internet.
+- Verifies the identity of cluster components and API users.
+
+Limitations
+===========
+
+Flocker's authentication layer does not completely guarantee the security of a cluster; it relies on the private keys of the cluster's certificate authority, control service, convergence agents and API users being kept secret.
+
+If, for example, a malicious user were able to gain root SSH access to the machine running the control service, they would be able to copy the control service's private key and therefore be able to set up another machine to act and identify as the legitimate control service for that cluster.
+Similarly, if the private key of an API end user is compromised, anyone with that key will be able to authenticate as that authorised user and therefore make requests to the REST API to read or change the state of a cluster.
+
+It is therefore very important that you ensure the private keys are kept secret; they should not be copied or shared.
```

Comment:
With the exception of copying the relevant private keys once over a secure medium to install them on a node, they shouldn't be copied or shared at all. What I want to make clear in this bit is "only the control service node should have a copy of the control service private key" - wording suggestions welcome.